code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -766,15 +767,20 @@
 	if(loc == src)
 		return
 	old_upgrade.deactivate(src)
-	upgrades -= old_upgrade
+	for(var/obj/item/borg/upgrade/current_upgrade in src.upgrades)
+		if(current_upgrade.name == old_upgrade.name)
```

Comment:
Why by name, if the passed in upgrade is not in the cyborg already there's something wrong with this whole system.

Reply:
you can not compare objects in this manner in this case, as different objects will have a variety of different variables, regardless of being the same type of object, as far as I could tell from testing.

Reply:
Because this current implementation doesn't work from what I saw in testing. When it attempts to remove the object, it will always fail.